src/api/dependency-discovery/src/github.js

```diff
@@ -0,0 +1,66 @@
+const { request } = require('@octokit/request');
+
+// Check whether a specific time has reached the GitHub time limit.
+// @param {Date} datetime - a Date object representing a moment in the past
+// @returns {Boolean}
+const hasGitHubExpired = (dateTime) => new Date() - dateTime > 60 * 60 * 1000;
```

Comment:
A better way to do this is with an LRU cache, for example https://www.npmjs.com/package/lru-cache.  It can handle the TTL stuff and expiring keys.

Can you file a follow-up with an issue to do this in Redis instead of in memory?  That way starting/stopping the containers won't lose the cache (e.g., when we re-deploy).  Redis can handle the TTL for you when you add a key, and we can stuff the entire JSON response into the value.

Reply:
@humphd, would you like me to switch the current implementation to a LRU, or would you prefer to leave it as it is and change it to Redis for 3.0-alpha?

Reply:
The issue is filed #3323.

Reply:
Perfect.

Reply:
Let's leave for now, and do Redis (which is the proper way).